crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -1635,7 +1636,7 @@ impl Locking {
         for write in &transaction.writes {
             let table_id = TableId(write.set_id);
             let schema = inner.schema_for_table(table_id)?;
-            let row_type = inner.row_type_for_table(table_id)?;
+            let row_type = inner.row_type_for_table(table_id)?.into_owned();
             let table = inner.committed_state.tables.entry(table_id).or_insert(Table {
                 row_type: row_type.clone(),
```

Comment:
It seems if we move `table` into the `match`, we could get rid of this clone as well. It's only used in `insert` operation and in a way where `table` can be constructed after it's used.